src/storage/txn/scheduler.rs

```diff
@@ -234,8 +244,11 @@ impl<L: LockManager> SchedulerInner<L> {
     fn acquire_lock(&self, cid: u64) -> bool {
         let mut task_contexts = self.task_contexts[id_index(cid)].lock();
         let tctx = task_contexts.get_mut(&cid).unwrap();
+        tctx.on_schedule();
         if self.latches.acquire(&mut tctx.lock, cid) {
-            tctx.on_schedule();
+            SCHED_LATCH_HISTOGRAM_VEC
+                .get(tctx.tag)
+                .observe(tctx.wait_timer.elapsed_secs());
```

Comment:
The timer is reset at L247, so the latch wait time is always smaller than `SCHED_WAIT`?

Reply:
This change is for fixing:
```
3. The SCHED_LATCH_HISTOGRAM_VEC include duration in queue and latch time, which is not correct
```
After this change, the SCHED_WAIT will response for `waiting for an available thread`, and SCHED_LATCH will response for `waiting for the latch`